ui/js/graphs.js

```diff
@@ -684,21 +688,23 @@ perf.controller('GraphsCtrl', [
             });
 
             modalInstance.result.then(function(series) {
-              series.highlightedPoints = [];
-              series.visible = true;
-              series.color = availableColors.pop();
-
-              $scope.seriesList.push(series);
-              if( !$scope.highlightedRevision ) {
-                $scope.highlightedRevision = '';
-              }
-              if (!$scope.zoom) {
-                $scope.zoom = {};
+              for (var i = 0; i < series.length; i++) {
+                  series[i].hightlightedPoints = [];
+                  series[i].visible = true;
+                  series[i].color = availableColors.pop();
+                  $scope.seriesList.push(series[i]);
+                  console.log($scope.seriesList);
+                  if (!$scope.highlightedRevision) {
+                      $scope.highlightedRevision = '';                  
+                  }
+                  if (!$scope.zoom) {
+                      $scope.zoom = {};
+                  }
+                  updateDocument();
+                  getSeriesData(series[i]).then(function() {
```

Comment:
So we only want to call plotGraph() after we've downloaded data for all the series selected. The reason you're getting an undefined warning here is because you're calling plotGraph() while we're still downloading data for a series. Using $q.all() here on the promises returned by getSeriesData should fix this problem.

For more info, see: https://docs.angularjs.org/api/ng/service/$q

Incidentally, this block of code could be improved in a few other ways-- I'll give more comments on Monday at the latest-- until then, hopefully this helps a bit. :)